crates/partition-store/src/invocation_status_table/mod.rs

```diff
@@ -151,18 +112,56 @@ fn get_invocation_status<S: StorageAccess>(
     // Try read the old one first
     // The underlying assumption is that an invocation status will never exist in both old and new
     // invocation status table.
-    if let Some(s) =
-        storage.get_value::<_, InvocationStatus>(create_invocation_status_key(invocation_id))?
+    if let Some(s) = storage
+        .get_value::<_, InvocationStatusV1>(create_invocation_status_key_v1(invocation_id))?
     {
-        return Ok(s);
+        return Ok(s.0);
     }
 
     // todo: Read first from new invocation status table once the runtime writes to this table
     storage
-        .get_value::<_, InvocationStatusV2>(create_neo_invocation_status_key(invocation_id))
+        .get_value::<_, InvocationStatus>(create_invocation_status_key(invocation_id))
+        .map(|value| {
+            if let Some(invocation_status) = value {
+                invocation_status
+            } else {
+                InvocationStatus::Free
+            }
+        })
```

Comment:
Is it possible to swap the order? First trying to read the new invocation status? I suspect that in the future the new invocation status will hit more often than hitting the old version.

Reply:
We can do it.

Another thing I would like to do is the cheap optimization of skipping all the "v1" codepaths when the v1 table is empty on partition store startup. Perhaps in a followup (together with the idempotency table skip followup)